src/app/authentication/sign-up/sign-up.component.ts

```diff
@@ -59,7 +59,7 @@ export class SignUpComponent implements OnInit {
                 usr['auth'] = SocialRes.authToken;
                 usr['provider'] = SocialRes.provider;
 
-                this.loginService.socialLogin(GoogleLoginProvider.PROVIDER_ID, SocialRes.authToken)
+                this.loginService.socialLogin(provider, SocialRes.authToken)
```

Comment:
Can you go over this entire method and see which things can be made better. We'll try to fix some thing by the way. If you don't see any, we'll do it together.